refarch-integrations/refarch-s3-integration/refarch-s3-integration-service/src/main/java/de/muenchen/refarch/s3/integration/security/RequestResponseLoggingFilter.java

```diff
@@ -0,0 +1,84 @@
+package de.muenchen.refarch.s3.integration.security;
+
+import jakarta.servlet.*;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.annotation.Order;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.stereotype.Component;
+
+/**
+ * This filter logs the username for requests.
+ */
+@Component
+@Order(1)
+@Slf4j
+public class RequestResponseLoggingFilter implements Filter {
+
+    private static final String REQUEST_LOGGING_MODE_ALL = "all";
+
+    private static final String REQUEST_LOGGING_MODE_CHANGING = "changing";
+
+    private static final List<String> CHANGING_METHODS = Arrays.asList("POST", "PUT", "PATCH", "DELETE");
+
+    /**
+     * The property or a zero length string if no property is available.
+     */
+    @Value("${security.logging.requests:}")
```

Comment:
Could be refactored into a `@ConfigurationProperties` annotated class and then autowired here.

Reply:
Refactoring to correct properties should be done withing #55